website/static/js/registrationModal.js

```diff
@@ -47,21 +45,31 @@ var RegistrationViewModel = function(confirm, prompts, validator) {
         }
     );
     self.embargoEndDate = ko.computed(function() {
-        return new Date(self.pikaday());
+        return moment(new Date(self.pikaday())).subtract(utcOffset, 'minutes');
     });
 
+    self.minimumTimeValidation = function (embargoLocalDateTime) {
+        var minEmbargoMoment = getMinimumDate(embargoLocalDateTime),
+            endEmbargoMoment = self.embargoEndDate();
+        return minEmbargoMoment.isBefore(endEmbargoMoment);
+    };
+
+    self.maximumTimeValidation = function (embargoLocalDateTime) {
+        var maxEmbargoMoment = getMaximumDate(embargoLocalDateTime),
+            endEmbargoMoment = self.embargoEndDate();
+        return maxEmbargoMoment.isAfter(endEmbargoMoment);
+    };
+
     var validation = [{
-        validator: function() {
-            var endEmbargoDateTimestamp = self.embargoEndDate().getTime();
-            return endEmbargoDateTimestamp > TWO_DAYS_FROM_TODAY_TIMESTAMP;
+        validator: function () {
+            return self.minimumTimeValidation();
         },
         message: 'Embargo end date must be at least three days in the future.'
     }, {
-	validator: function() {
-            var endEmbargoDateTimestamp = self.embargoEndDate().getTime();
-	    return endEmbargoDateTimestamp < FOUR_YEARS_FROM_TODAY_TIMESTAMP;
-	},
-	message: 'Embargo end date must be less than four years in the future.'
+        validator: function () {
```

Comment:
And here. The advantage of this is that, when you're debugging, you don't have an anonymous function in your call stack. It makes things much easier.